src/trigger/scripttrigger.cpp

```diff
@@ -94,7 +100,12 @@ ScriptTrigger::event(Player& , EventType type)
   if(type != triggerevent)
     return;
 
+  if (oneshot && runcount >= 1) {
```

Comment:
If we're only ever having two conditions (runcount 1 or 2), why not turn this into a bool?

Reply:
Because, a) maybe in the future we could expose the run count to the script, even though I don't know why this would be useful, and b) because I found it easier to increment a counter. Also, the states are not 1 or 2, but essentially 0 (not triggered) or 1 (triggered, which now means disable).

Reply:
Maybe you should move the 1 into a private max_runcount variable? Increases visibility and makes it easier to expose to scripting later on.